packages/forms-web-app/src/dynamic-forms/journey.js

```diff
@@ -206,14 +206,14 @@ class Journey {
 					reverse ? j-- : j++
 				) {
 					const question = currentSection.questions[j];
-					if (takeNextQuestion) {
+					if (takeNextQuestion && question.shouldDisplay(this.response)) {
```

Comment:
getStatus of section will probably need to handle question.shouldDisplay as well I think?
trying to think of anywhere else it may be needed

Reply:
Ok that is a new concept I wasn't aware of

Reply:
I've stuck a condition in the getStatus method, lemme know if you think I've done the right thing there. WRT the looping I'm wondering if either sections or journeys should have a method to get all the relevant questions ie omitting ones that aren't displayed

Reply:
yes it should be that simple and the section completion logic should now be correct
A single method could certainly centralize that logic rather than needing to remember to do it everywhere!